src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -87,74 +87,57 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
         }
 
 #if SYSTEM_PRIVATE_CORELIB
+        // Converts Vector128<byte> into 2xVector128<byte> ASCII Hex representation
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static (Vector128<byte>, Vector128<byte>) AsciiToHexVector128(Vector128<byte> src, Vector128<byte> hexMap)
+        {
+            Debug.Assert(Ssse3.IsSupported || AdvSimd.Arm64.IsSupported);
+            // The algorithm is simple: a single srcVec (contains the whole 16b Guid) is converted
+            // into nibbles and then, via hexMap, converted into a HEX representation via
+            // Shuffle(nibbles, srcVec). ASCII is then expanded to UTF-16.
+            Vector128<byte> nibbles = Vector128.ShiftRightLogical(src.AsUInt64(), 4).AsByte();
+            Vector128<byte> lowNibbles = Vector128.UnpackLow(nibbles, src);
+            Vector128<byte> highNibbles = Vector128.UnpackHigh(nibbles, src);
+
+            return (ShuffleUnsafe(hexMap, lowNibbles & Vector128.Create((byte)0xF)),
+                ShuffleUnsafe(hexMap, highNibbles & Vector128.Create((byte)0xF)));
+
+            // TODO: remove once https://github.com/dotnet/runtime/pull/80963 is merged
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static Vector128<byte> ShuffleUnsafe(Vector128<byte> value, Vector128<byte> mask)
+                => Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
+        }
+
         private static void EncodeToUtf16_Vector128(ReadOnlySpan<byte> bytes, Span<char> chars, Casing casing)
         {
-            Vector128<byte> shuffleMask = Vector128.Create(
-                0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
-                0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF);
-
-            Vector128<byte> asciiTable = (casing == Casing.Upper) ?
-                Vector128.Create((byte)'0', (byte)'1', (byte)'2', (byte)'3',
-                                 (byte)'4', (byte)'5', (byte)'6', (byte)'7',
-                                 (byte)'8', (byte)'9', (byte)'A', (byte)'B',
-                                 (byte)'C', (byte)'D', (byte)'E', (byte)'F') :
-                Vector128.Create((byte)'0', (byte)'1', (byte)'2', (byte)'3',
-                                 (byte)'4', (byte)'5', (byte)'6', (byte)'7',
-                                 (byte)'8', (byte)'9', (byte)'a', (byte)'b',
-                                 (byte)'c', (byte)'d', (byte)'e', (byte)'f');
+            Debug.Assert(bytes.Length >= Vector128<int>.Count);
 
-            nuint pos = 0;
-            Debug.Assert(bytes.Length >= 4);
+            ref byte srcRef = ref MemoryMarshal.GetReference(bytes);
+            ref ushort destRef = ref Unsafe.As<char, ushort>(ref MemoryMarshal.GetReference(chars));
+
+            Vector128<byte> hexMap = casing == Casing.Upper ?
+                Vector128.Create("0123456789ABCDEF"u8) :
+                Vector128.Create("0123456789abcdef"u8);
```

Comment:
What are the before/after Guid formatting benchmark results on Mono?

Reply:
Ah, oops, ` Vector128.Create("h0e0l0l0o0 0w00"u8);` is not correct (input's length is not 16) and Mono emitted `ThrowInvalidException` for it correctly, let me re-run it for a correct input

Reply:
> Are you concerned about the codegen for this on Mono?

Yes.

We had a fire drill at the end of .NET 7 where vectorization introduced major ship-stopped regressions on Mono.

The RCA (root cause analysis) report produced for this incident had number of action items to prevent the repeat of the situation. I can forward the report if you have not seen it. One of the action item was: "Libraries work that exercises new runtime features or patterns will be tested in more scenarios during implementation". This is using a new runtime feature, so I am asking to validate the perf impact on Mono according to what we said that we are going to do in the RCA.

cc @jeffhandley @adamsitnik @DrewScoggins @lewing @sblom

Reply:
@jkotas thanks! I've reverted the change, although, I feel like even with sub-optimal codegen it should still be faster than .NET 7.0, e.g. here are the wins from the initial vectorization in CoreCLR: 

Also, it looks like we don't run benchmarks for LLVM JIT/AOT, do we? I don't see anything here https://aka.ms/HistoryIndex/ Presumably, WASM is interp there and Ubuntu Mono AOT aren't LLVM judging by the configs. So looks like Vector128 APIs aren't benchmarked at all since regular Mono (mini) doesn't yet support them.

Reply:
@DrewScoggins @lewing @sblom Can you comment on benchmarks that are run on Mono?

Reply:
On wasm SIMD still isn't supported on all the major browsers (Safari just enabled in the tech preview) so none of the vector paths are currently benchmarked in dotnet/performance. We have a few tiny benchmarks we run ourselves while we work on it.  I'm not sure offhand which scenarios are benchmarked for mono on the other platforms but @SamMonoRT and @steveisok should be able to fill in the details.

Reply:
Mono AOT-LLVM benchmarks should be running in the perf labs -- There has always been a problem with the labels where it keeps saying LLVM==false, but in fact it is enabled. @DrewScoggins @naricc should be able to confirm that.

Reply:
Ah I see, so the improvement from https://github.com/dotnet/runtime/pull/81650 is expected to show up here: https://pvscmdupload.blob.core.windows.net/reports/allTestHistory%2frefs%2fheads%2fmain_arm64_ubuntu%2020.04_LLVM%3dfalse_MonoAOT%3dtrue_MonoInterpreter%3dfalse_RunKind%3dmicro_mono%2fSystem.Tests.Perf_Guid.GuidToString.html (the last data point is 4th of Feb).

Reply:
@EgorBo Yes.

Reply:
@EgorBo - there seems to be some reporting issues in the perf labs, which is not showing results from past 2 days. @LoopedBard3 is looking into it and we'll see if there is a way to get you reports manually.

Reply:
> the data is updated but it didn't improve nor regressed from https://github.com/dotnet/runtime/pull/81650 - it doesn't make sense to me unless AdvSimd.Arm64.IsSupported returned false

Is that a property of the runtime or something related to the physical hardware that the run is happening on? Both the CoreCLR and Mono runs are happening on the same hardware, so I would not expect one to have this property be true and another be false. It seems useful to track down why we are not seeing this win when we expect there to be one.

Reply:
Yeah, I am 100% sure it either doesn't use LLVM actually or `AdvSimd.Arm64.IsSupported` is never true there. There is no way that benchmark wouldn't be affected otherwise